server/game/cards/25.7-MoM/MaesterJoseran.js

```diff
@@ -0,0 +1,24 @@
+const DrawCard = require('../../drawcard');
+const GameActions = require('../../GameActions');
+
+class MaesterJoseran extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                onFirstPlayerDetermined: event =>
+                    this.controller === event.player &&
+                    this.game.getPlayers().every(player => player.agenda)
+            },
+            message: '{player} uses {source} to have each player put the top card of their decks under their agenda',
+            handler: context => {
+                const topCards = this.game.getPlayers().map(player => player.drawDeck.slice(0, 1));
+                this.game.resolveGameAction(GameActions.simultaneously(() => topCards.map(card => GameActions.placeCard({ player: card.controller, card, location: 'conclave' }))), context);
+            }

```

Comment:
This can probably just be a `gameAction` (more for knowledge rather than holding up the implementation)

```suggestion
            gameAction: GameActions.simultaneously(context =>
                context.game.getPlayers().map(player => player.drawDeck.slice(0, 1)).map(card => 
                    GameActions.placeCard({ player: card.controller, card, location: 'conclave' })
                )
            )
```